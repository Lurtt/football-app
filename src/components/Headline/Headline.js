import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { textAlign, letterSpacing, margin, padding } from '../mixins'

export default function Headline({
  level = 1,
  color = '#444',
  children,
  ...props
}) {
  const HeadlineElement = styled(`h${level}`)`
    color: ${color};
    ${margin};
    ${padding};
    ${letterSpacing};
    ${textAlign};
    text-transform: ${({ uppercase }) => (uppercase ? 'uppercase' : 'none')};
  `

  return <HeadlineElement {...props}>{children}</HeadlineElement>
}

Headline.propTypes = {
  level: PropTypes.number,
  textAlign: PropTypes.oneOf(['left', 'center', 'right']),
  color: PropTypes.string,
  letterSpacing: PropTypes.string,
  margin: PropTypes.string,
  padding: PropTypes.string,
  children: PropTypes.node,
}
