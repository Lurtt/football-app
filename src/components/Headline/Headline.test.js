import React from 'react'
import renderer from 'react-test-renderer'

import Headline from './'

describe('<Headline />', () => {
  test('Should render h1 by default', () => {
    const result = renderer.create(<Headline>Headline h1</Headline>).toJSON()
    expect(result).toMatchSnapshot()
  })
  test('Should render h2', () => {
    const result = renderer
      .create(<Headline level={2}>Headline h2</Headline>)
      .toJSON()
    expect(result).toMatchSnapshot()
  })
})
