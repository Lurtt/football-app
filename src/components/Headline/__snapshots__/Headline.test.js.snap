// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Headline /> Should render h1 by default 1`] = `
<h1
  className="sc-bdVaJa cZJVNZ"
>
  Headline h1
</h1>
`;

exports[`<Headline /> Should render h2 1`] = `
<h2
  className="sc-bwzfXH qoDLU"
>
  Headline h2
</h2>
`;
